src/tabs.ts

```diff
@@ -0,0 +1,258 @@
+import {LitElement, html} from 'lit';
+import {customElement, state, property} from 'lit/decorators.js';
+import {animate_defs, animationCSS, ResponsiveVal,
+        getResponsiveValues, parseAttrVal} from "./utils";
+import {styleMap} from 'lit/directives/style-map.js';
+
+@customElement('aalam-tabs')
+export class AalamTabs extends LitElement {
+    DEFAULT_VALUES:{ [key:string]: any } = {
+        animationDur: 100,
+        activecls: 'tab-active',
+        fashion: 'xs:row',
+        colsize: 'title:30%;body:70%',
+        internal_fashion: 'row',
+        fashion_style: {xs:'row'},
+        column_size:{title:'30%', body:'70%'}
+    }
+
+    @property({type:String, attribute:true})
+    animation = '';
+
+    @property({type:Number, attribute:true})
+    animationDur = this.DEFAULT_VALUES.animationDur;
+
+    @property({type:String, attribute:true})
+    activecls = this.DEFAULT_VALUES.activecls;
+
+    @property({type:String, attribute:true})
+    fashion = this.DEFAULT_VALUES.fashion;
+
+    @property({type:String, attribute:true})
+    colsize = this.DEFAULT_VALUES.colsize;
+
+    @state()
+    _internal_fashion:string = this.DEFAULT_VALUES.internal_fashion;
+
+    private _prev_ix:number;
```

Comment:
We can name this variable as _cur_ix as it represents the currently open iitem index